auth/oidc.go

```diff
@@ -212,73 +238,148 @@ func (op *OIDCProvider) DiscoverConfig() (config *oidc.ProviderConfig, shouldSyn
 	return config, shouldSync, err
 }
 
-func (op *OIDCProvider) FetchCustomProviderConfig(discoveryURL string) (*oidc.ProviderConfig, error) {
-
-	var customConfig OidcProviderConfiguration
+func GetOIDCUsername(provider *OIDCProvider, subject string) string {
+	return fmt.Sprintf("%s_%s", provider.UserPrefix, url.QueryEscape(subject))
+}
 
+// Clone of provider metadata construction from go-oidc library.
+// https://github.com/coreos/go-oidc/blob/v2/oidc.go#L113
+func (op *OIDCProvider) FetchCustomProviderConfig(ctx context.Context, discoveryURL string) (*oidc.Provider, error) {
 	// If discovery URL is empty, use the standard discovery URL
 	if discoveryURL == "" {
```

Comment:
We're already doing this check in DiscoverConfig.  I think we should return error here if an empty string is passed in, instead of calling back to standard.